### PR TITLE
Major refactoring of the install procedure

### PR DESCRIPTION
**Pull request by atosatto**
Major refactoring of the tasks adding the PowerDNS official repositories.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,72 +1,121 @@
...
 
 Role Variables
 --------------
-## dnsdist_acls
+
+Available variables are listed below, along with default values (see `defaults/main.yml`):
+
+    dnsdist_install_repo: False
+
+By default the PowerDNS Recursor is installed from the os default repositories.
```
Comment (pieterlexis) [inline README.md]: s/PowerDNS Recursor/dnsdist/

File: README.md [hunk 2]
```diff
@@ -1,72 +1,121 @@
...
+You can install the PowerDNS Recursor package from official PowerDNS repository
```
Comment (pieterlexis) [inline README.md]: s/PowerDNS Recursor/dnsdist/

File: README.md [hunk 3]
```diff
@@ -1,72 +1,121 @@
...
+overriding the `dnsdist_install_repo` variable value as follows:
+
+    # Install dnsdist from the master branch
+    - hosts: pdns-dnsdists
+      roles:
+      - { role: PowerDNS.dnsdist,
+          dnsdist_install_repo: "{{ dnsdist_official_dnsdist_master }}"
```
Comment (pieterlexis) [inline README.md]: 'official' sounds very...... uhm.. official.. I don't have a better suggestion though

File: tasks/main.yml [hunk 4]
```diff
@@ -1,13 +1,36 @@
...
+    name: dnsdist
+    state: present
+  tags:
+    - install
+
+- name: Add the dnsdist configuration
+  template:
+    src: dnsdist.conf.j2
+    dest: /etc/dnsdist/dnsdist.conf
+    validate: 'dnsdist -C %s --check-config 2>&1'
```
Comment (pieterlexis) [inline tasks/main.yml]: with this line here, the validation in the handler can go

File: tasks/main.yml [hunk 5]
```diff
@@ -1,13 +1,36 @@
...
-- include: "install.packages-{{ansible_os_family}}.yml"
+
+- include: "repo-{{ ansible_os_family }}.yml"
+  when: dnsdist_install_repo
+  tags:
+    - install
+    - repository
+
+- include: install-epel.yml
+  when: ansible_os_family == 'RedHat'
```
Comment (pieterlexis) [inline tasks/main.yml]: maybe we should have a `dnsdist_install_epel` variable to support people that have private EPEL-like repos?

File: README.md (hunk 3 above)
Comment (atosatto) [inline README.md]: Switched to `dnsdist_pdns_dnsdist_master`.

File: tasks/main.yml (hunk 5 above)
Comment (atosatto) [inline tasks/main.yml]: Good hint! ;-)

File: tasks/main.yml (hunk 4 above)
Comment (atosatto) [inline tasks/main.yml]: Removed in @806f75a

**Comment by pieterlexis** [comment]
Please squash your changes before merging :+1:

**Pull request merged**